lib/puppet/type/vm_vapp_property.rb

```diff
@@ -17,7 +17,11 @@
     desc 'The datacenter, vm name and visible label of the property split by a colon (:). Format dc1:vm1:label'
 
     munge do |value|
-      @resource[:datacenter], @resource[:vm_name], @resource[:label] = value.split(':',3)
+      if value =~ /^[a-zA-Z0-9\-\. ]+:[a-zA-Z0-9\-\. ]+:[a-zA-Z0-9\-\. ]+$/ then
```

Comment:
Or maybe make it even dumber than that (but more inline with the original code) and just do the split if there are two colons present....

```ruby
if value.count(':') == 2
  value.split(.....)
end
```